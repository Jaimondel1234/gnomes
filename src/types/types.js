export const types = {
  login: "[Auth] Login",
  logout: "[Auth] Logout",
  loginRequest: "[Auth] Login Request",
};
