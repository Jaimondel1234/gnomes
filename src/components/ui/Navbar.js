import React from "react";
import { Link } from "react-router-dom";
import { useDispatch } from "react-redux";
import { logout } from "../../actions/auth";
import { NavbarList, NavbarElement, ButtonStyle } from "../../styles/core";
import styled from "styled-components";

const ButtonLogoutStyle = styled(ButtonStyle)`
  display: block;
  font-weight: bold;
  font-size: 40px;
  text-align: center;
`;

export const Navbar = () => {
  const dispatch = useDispatch();

  const handleLogout = () => {
    dispatch(logout());
    localStorage.removeItem("user");
  };

  return (
    <>
      <NavbarList>
        <NavbarElement>
          <Link exact to="/home">
            Home
          </Link>
        </NavbarElement>
        <NavbarElement>
          <Link exact to="/search">
            Search
          </Link>
        </NavbarElement>
        <NavbarElement>
          <Link exact to="/all-gnomes">
            Gnomes
          </Link>
        </NavbarElement>
        <NavbarElement float="right">
          <ButtonLogoutStyle onClick={handleLogout} m="5px" p="5px 10px">
            Logout
          </ButtonLogoutStyle>
        </NavbarElement>
      </NavbarList>
    </>
  );
};
