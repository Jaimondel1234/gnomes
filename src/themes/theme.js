/* 
  Theme
*/

export const theme = {
  color: {
    main: '#007bff',
  },
};
