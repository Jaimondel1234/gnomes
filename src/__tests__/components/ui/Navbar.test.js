import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import { theme } from '../../../themes/theme';
import { Navbar } from '../../../components/ui/Navbar';
import { MemoryRouter } from 'react-router-dom';
import { types } from '../../../types/types';

describe('Test <Navbar/>', () => {
  Storage.prototype.removeItem = jest.fn();
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initState = {};

  const store = mockStore(initState);
  store.dispatch = jest.fn();
  const wrapper = mount(
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <MemoryRouter>
          <Navbar />
        </MemoryRouter>
      </ThemeProvider>
    </Provider>,
  );
  test('it must be displayed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('if the button is clicked, the logout action must be dispatched', () => {
    wrapper.find('button').prop('onClick')();
    expect(localStorage.removeItem).toHaveBeenCalledWith('user');
    expect(store.dispatch).toHaveBeenCalledWith({ type: types.logout });
  });
});
