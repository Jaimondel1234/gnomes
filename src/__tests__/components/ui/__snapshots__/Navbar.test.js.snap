// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Navbar/> it must be displayed correctly 1`] = `
<ul
  className="core__NavbarList-qu5lcc-10 IJNH"
>
  <li
    className="core__NavbarElement-qu5lcc-11 llrJN"
  >
    <a
      href="/home"
      onClick={[Function]}
    >
      Home
    </a>
  </li>
  <li
    className="core__NavbarElement-qu5lcc-11 llrJN"
  >
    <a
      href="/search"
      onClick={[Function]}
    >
      Search
    </a>
  </li>
  <li
    className="core__NavbarElement-qu5lcc-11 llrJN"
  >
    <a
      href="/all-gnomes"
      onClick={[Function]}
    >
      Gnomes
    </a>
  </li>
  <li
    className="core__NavbarElement-qu5lcc-11 bAOUru"
  >
    <button
      className="core__ButtonStyle-qu5lcc-6 Navbar__ButtonLogoutStyle-sc-1p7v5fe-0 eXWduD fgIWp"
      onClick={[Function]}
    >
      Logout
    </button>
  </li>
</ul>
`;
